File: materials/2016-2017/11b/2016-12-08/Circle.cc
#include "Point.hh"
#include "Circle.hh"
#include <iostream>

Circle::Circle(Point& center, double radius)
	: center_(center), radius_(radius) {}

void Circle::draw() const {
	std::cout << "Circle(center=(" << center_.get_x() << ","
			  << center_.get_y() << "),radius=" << radius_ << ")"
			  << std::endl;
}

File: materials/2019-2020/11b/2020-02-26-java-testing-exercise/src/main/java/com/oop/Circle.java
package main.java.com.oop;

public class Circle {
    private final Point center;
    private final double radius;

    public double getRadius() {
        return radius;
    }

    public Circle(Point center, double radius) {
        this.center = center;
        this.radius = radius;
    }

    public double getPerimeter() {
        return 2 * Math.PI * radius;
    }

    public double getArea() {
        return Math.PI * Math.pow(radius, 2);
    }

    public void printShape() {
        System.out.println("Circle");
        System.out.println("Center = (" + center.getX() + ", " + center.getY() + ")");
        System.out.println("Radius = " + radius);
        System.out.println();
        System.out.println("Perimeter = " + this.getPerimeter());
        System.out.println("Area = " + this.getArea());
    }
}
